apps/dsb-message-broker/src/channel/channel.service.ts

```diff
@@ -22,42 +21,47 @@ export class ChannelService implements OnModuleInit {
         });
     }
 
-    public async create(
+    public async createChannel(
         channelData: CreateChannelDto & { createdBy: string; createdDateTime: string }
     ): Promise<string> {
-        const { org, app, channel } = extractFqcn(channelData.fqcn);
-        const organizations = this.configService.get('ORGANIZATIONS');
-        const fqcnIsValid = organizations.some((_org: any) => {
-            if (_org.name !== org) return false;
-            return _org.apps.some((_app: any) => {
-                if (_app.name !== app) return false;
-
-                return new RegExp(_app.channels).test(channel);
-            });
-        });
-        if (!fqcnIsValid) throw new Error('fqcn does not match the defined pattern.');
+        if (!channelData.admins) channelData.admins = [channelData.createdBy];
 
         return this.transport.createChannel(channelData);
     }
 
-    public async getAvailableChannels(userDID: string, userVR: string[]): Promise<Channel[]> {
+    public async updateChannel(
+        channelData: CreateChannelDto & { modifiedBy: string; modifiedDateTime: string }
```

Comment:
I think it's worth stating this is a `Partial<CreateChannelDto>` (given it's a PATCH) - that also means fqcn could be missing (even though it is required to update the channel.

Reply:
other fields than fqcn is optional in dto and i think it implies that partial expectation.(forgot to change it to UpdateChannel Dto)